### PR TITLE
Feature/orb 6 selected products

### PR DESCRIPTION
**Pull request by brimorales**
Selected products feature.

Captura de Pantalla 2021-08-04 a la(s) 22 14 47

### CODE REVIEW

File: app/assets/stylesheets/product.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Product controller here.
```
Comment (diebas) [inline app/assets/stylesheets/product.scss]: remove unused files

File: app/helpers/articles_helper.rb [hunk 2]
```diff
@@ -1,2 +1,4 @@
+# frozen_string_literal: true
```
Comment (diebas) [inline app/helpers/articles_helper.rb]: remove al this empty helpers

File: app/javascript/packs/script.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  }
+})
+document.querySelector(".minus-btn").addEventListener("click", function(){
+  valueCount = document.getElementById("quantity").value;
+  valueCount--;
+  document.getElementById("quantity").value = valueCount;
+  if (valueCount == 1) {
+    document.querySelector(".minus-btn").setAttribute("disabled", "disabled");
+  }
+})
```
Comment (diebas) [inline app/javascript/packs/script.js]: add blank line at the end of file

File: app/channels/application_cable/channel.rb [hunk 4]
```diff
@@ -1,3 +1,5 @@
+# frozen_string_literal: true
```
Comment (diebas) [inline app/channels/application_cable/channel.rb]: it's not necessary to add this comment in every file, just add it where used

File: app/models/article.rb [hunk 5]
```diff
@@ -1,17 +1,20 @@
+# frozen_string_literal: true
```
Comment (diebas) [inline app/models/article.rb]: why are you adding articles?

**Review by diebas — COMMENTED** [review]
watch out here, you're adding a lot of unintended changes

File: bin/bundle [hunk 6]
```diff
@@ -8,62 +8,64 @@
...
 
-require "rubygems"
+require 'rubygems'
 
 m = Module.new do
   module_function
 
   def invoked_as_script?
-    File.expand_path($0) == File.expand_path(__FILE__)
+    File.expand_path($PROGRAM_NAME) == File.expand_path(__FILE__)
```
Comment (diebas) [inline bin/bundle]: what are this changes for?

File: config/routes.rb [hunk 7]
```diff
@@ -1,27 +1,12 @@
...
-  get "bienvenida", to: "home#index"
-  get "perfil", to: "users#edit"
-
-  resources :articles do
-    get "/user/:user_id", to: "articles#from_author", on: :collection
-  end
+  root to: 'home#index'
+  get 'bienvenida', to: 'home#index'
+  get 'perfil', to: 'users#edit'
   resources :categories
```
Comment (diebas) [inline config/routes.rb:8]: are you using this route?

File: config/routes.rb [hunk 8]
```diff
@@ -1,27 +1,12 @@
...
   # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
-  root to: "home#index"
-  get "bienvenida", to: "home#index"
-  get "perfil", to: "users#edit"
-
-  resources :articles do
-    get "/user/:user_id", to: "articles#from_author", on: :collection
-  end
+  root to: 'home#index'
+  get 'bienvenida', to: 'home#index'
```
Comment (diebas) [inline config/routes.rb]: use english words for routes as the app is in english

File: test/controllers/articles_controller_test.rb [hunk 9]
```diff
@@ -1,4 +1,6 @@
-require "test_helper"
+# frozen_string_literal: true
+
+require 'test_helper'
 
 class ArticlesControllerTest < ActionDispatch::IntegrationTest
```
Comment (diebas) [inline test/controllers/articles_controller_test.rb]: remove this file

File: test/controllers/categories_controller_test.rb [hunk 10]
```diff
@@ -1,44 +1,46 @@
...
+
+require 'test_helper'
 
 class CategoriesControllerTest < ActionDispatch::IntegrationTest
   setup do
     @category = categories(:one)
   end
 
-  test "should get index" do
+  test 'should get index' do
```
Comment (diebas) [inline test/controllers/categories_controller_test.rb]: remove this file

File: test/models/article_test.rb [hunk 11]
```diff
@@ -1,4 +1,6 @@
-require "test_helper"
+# frozen_string_literal: true
+
+require 'test_helper'
```
Comment (diebas) [inline test/models/article_test.rb]: remove this file

File: test/models/category_test.rb [hunk 12]
```diff
@@ -1,4 +1,6 @@
-require "test_helper"
+# frozen_string_literal: true
+
+require 'test_helper'
```
Comment (diebas) [inline test/models/category_test.rb]: remove this file

File: test/models/has_category_test.rb [hunk 13]
```diff
@@ -1,4 +1,6 @@
-require "test_helper"
+# frozen_string_literal: true
+
+require 'test_helper'
```
Comment (diebas) [inline test/models/has_category_test.rb]: remove this file

File: test/system/categories_test.rb [hunk 14]
```diff
@@ -1,45 +1,47 @@
...
+
+require 'application_system_test_case'
 
 class CategoriesTest < ApplicationSystemTestCase
   setup do
     @category = categories(:one)
   end
 
-  test "visiting the index" do
+  test 'visiting the index' do
```
Comment (diebas) [inline test/system/categories_test.rb]: remove this file

File: bin/bundle (hunk 6 above)
Comment (brimorales) [reply bin/bundle]: That was one of Rubocop's changes, I guess it is something related to code's quality, isn't it?

File: app/views/products/index.html.erb [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+            </div>
+        <% end %>
+    <%end%>
+</div>
+<div class="pagy">
+<%== pagy_bootstrap_nav(@pagy) %>
+</div>
+
+
+
```
Comment (diebas) [inline app/views/products/index.html.erb]: too many lines at the end of the file

File: app/views/products/show.html.erb [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+          <li class="nav-item" style="list-style-type: none;">
+          <li class="nav-item" style="list-style-type: none;">
+            <%= link_to "Add to Cart", new_user_session_path, class:"btn btn-primary-sign-up btn-add-to-cart" %>
+            <%= link_to image_tag("carrito-compras-icon.png", class:"add-to-cart-icon") %>
+          </li>
+
+    </div>
+</div>
+<%= javascript_pack_tag 'script' %>
+<%= render "partials/footer" %>
```
Comment (diebas) [inline app/views/products/show.html.erb:27]: add blank line at the and of the file

**Review by diebas — APPROVED** [review]

**Pull request merged**